growth/too/views.py

```diff
@@ -1233,52 +1233,103 @@ def get_json_data(plan, decam_style=True):
     else:
         ditherNorm = 1.0
 
-    start_mjd = time.Time(plan.start_observation, format='datetime').mjd
-    end_mjd = time.Time(plan.end_observation, format='datetime')
-
-    # add a little buffer
-    end_mjd = (end_mjd + 30.0 * u.min).mjd
-
-    if doReferences:
-        json_data = {
-            'queue_name': "ToO_"+queue_name,
-            'validity_window_mjd': [start_mjd, end_mjd],
-            'targets': [
-                {
-                    'request_id': ii,
-                    'program_id': plan.program_id,
-                    'field_id': exposure.field_id,
-                    'ra': exposure.field.ra,
-                    'dec': exposure.field.dec,
-                    'filter_id': exposure.filter_id,
-                    'exposure_time': exposure.exposure_time/ditherNorm,
-                    'program_pi': program_pis[telescope],
-                    'subprogram_name': "ToO_"+transient_name
-                }
-                for ii, exposure in enumerate(exposures)
-                if exposure.filter_id in
-                exposure.field.reference_filter_ids
-            ]
-        }
+    if telescope == "ZTF":
```

Comment:
Why is this specific to ZTF?

Reply:
ZTF is the only one that doesnt receive the time of observations in the files to use, so it is what it is ;)

Reply:
Hmm.  Is it possible to at least factor out the common code to reduce duplication?

I get really nervous about having very different code paths for triggering different facilities.

Reply:
Hmmm the splitting is pretty particular... maybe @shreyasahasram08 has ideas?